### PR TITLE
Run3-hcx217 Move remaining geometry parameters for SIM step to HcalSimulationParameters

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

In the process of removing dependence of CompactView in the SIM step of HCAL, a few more variables are extracted and put in the object HcalSimulationParameters

#### PR validation:

Tested with standard runTheMatrix workflows

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CondFormats/GeometryObjects/interface/HcalSimulationParameters.h [hunk 1]
```diff
@@ -18,6 +18,14 @@ class HcalSimulationParameters {
   std::vector<int> pmtLeft_;
   std::vector<int> pmtFiberLeft_;
 
+  std::vector<int> hfLevels_;
```
Comment (ggovi) [inline CondFormats/GeometryObjects/interface/HcalSimulationParameters.h]: @bsunanda 
please make sure the evolution of  the CondFormats class  ( persistent type ) is limited to the pre* series. Once we have the 11_0_0, we won't be able to change the class any longer.

**Comment by bsunanda** [comment]
@ggovi OK. I do not foresee to change it further. This covers all the usage of HCAL as foreseen in simulation till date

**Comment by cvuosalo** [comment]
+1

**Comment by bsunanda** [comment]
@ggovi @christopheralanwest @tocheng @tlampen @pohsun Please check and sign this PR

**Comment by pohsun** [comment]
+1

**Comment by ggovi** [comment]
+1

**Pull request merged**